FilePersistence/src/version_control/Commit.cpp

```diff
@@ -103,56 +103,86 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 		return false;
 }
 
-QStringList Commit::nodeLinesFromId(Model::NodeIdType id, bool findChildrenByParentId) const
+bool Commit::isValidMatch(QString content, int indexOfId, int& start, int& end, bool findChildrenByParentId) const
 {
-	auto idtext =  id.toString();
-	QStringList matches;
-	for (auto file : files())
+	start = indexOfId;
+	end = indexOfId;
+
+	// start is the first character of the line containing id
+	while (start >= 0 && content[start] != '\n')
 	{
-		int indexOfId = 0;
-		auto content = QString::fromUtf8(file->content());
+		// String is of the form {.*} {id}
+		if (findChildrenByParentId == 0 && content[start] == '}')	return 0;
+		start--;
+	}
+	start++;
 
-		indexOfId = content.indexOf(idtext, indexOfId);
-		if (indexOfId != -1)
-		{
-			bool invalid = false;
-			auto start = indexOfId;
-			auto end = indexOfId;
+	// end is the character after the line containing id
+	while (end <= content.size() && content[end] != '\n')
+	{
+		// String is of the form {id} {*.}
+		if (findChildrenByParentId == 1 && content[end] == '{')	return 0;
+		end++;
+	}
+	return 1;
+}
 
-			// start is the first character of the line containing id
-			while (start >= 0 && content[start] != '\n')
-			{
-				// String is of the form {.*} {id}
-				if (!findChildrenByParentId)
-					if (content[start] == '}')
+QStringList Commit::nodeLinesFromId(Model::NodeIdType Id, QString workDir_, bool isWorkingDir,
```

Comment:
So, actually modifying this function is not the correct thing to do. Right now, the only place in Envision where we create objects of type `Commit` is `GitRepository::getCommit`. That one method explicitly prevents commits from being created from the working directory (the first `Q_ASSERT`). The correct thing to do is to modify  `GitRepository::getCommit` to also be able to return `Commit` objects that correspond to the current working directory. Then the code here should be almost untouched.

In `GitRepository::getCommit` you need to make an explicit test whether the input is a working directory and if so walk the files similarly to how you do here.